x-pack/plugins/fleet/server/collectors/agent_collectors.ts

```diff
@@ -81,52 +81,58 @@ export const getAgentData = async (esClient?: ElasticsearchClient): Promise<Agen
         },
         { error: 0, degraded: 0 }
       );
-    const response = await esClient.search({
-      index: AGENTS_INDEX,
-      size: 0,
-      aggs: {
-        versions: {
-          terms: { field: 'agent.version' },
-        },
-        last_checkin_status: {
-          terms: { field: 'last_checkin_status' },
+    const response = await esClient.search(
+      {
+        index: AGENTS_INDEX,
+        size: 0,
+        aggs: {
+          versions: {
+            terms: { field: 'agent.version' },
+          },
+          last_checkin_status: {
+            terms: { field: 'last_checkin_status' },
+          },
         },
       },
-    });
+      { signal: abortController.signal }
```

Comment:
[abortController](https://github.com/elastic/kibana/pull/144494#discussion_r1012965417) helps to terminate queries if the 1m task timeout is reached. Didn't pass it to all es queries yet as it requires changes in many files.

Currently the task takes about 300m to run on average, will keep an eye on it after adding the agent logs queries.

Reply:
Thanks for being mindful about performance here. As I understand it, things like background tasks, telemetry collection, etc will eventually run on Kibana's separate "service" process, so heavy operations like this will eventually be optimized such that they aren't taking time or resources away from the main Kibana server/UI. Still, it's good to make sure we're not running a bunch of long running queries here.